modules/seastate/src/SeaState.f90

```diff
@@ -325,31 +255,22 @@ SUBROUTINE SeaSt_Init( InitInp, u, p, x, xd, z, OtherState, y, m, Interval, Init
    
       ! Define initialization-routine output here:
       InitOut%Ver = SeaSt_ProgDesc         
-      ! These three come directly from processing the inputs, and so will exist even if not using Morison elements:
  
-      CALL SeaStOut_Init( SeaSt_ProgDesc, InitInp%OutRootName, InputFileData, y,  p, m, InitOut, ErrStat2, ErrMsg2 ); CALL SetErrStat(ErrStat2,ErrMsg2,ErrStat,ErrMsg,RoutineName)
-         IF ( ErrStat >= AbortErrLev ) THEN
-            CALL CleanUp()
-            RETURN
-         END IF
-
-!===============================================
+      CALL SeaStOut_Init( SeaSt_ProgDesc, InitInp%OutRootName, InputFileData, y,  p, m, InitOut, ErrStat2, ErrMsg2 ); if(Failed()) return;
           
-      CALL SeaStOut_WrSummaryFile(InitInp, InputFileData, p, ErrStat2, ErrMsg2)
-         CALL SetErrStat(ErrStat2,ErrMsg2,ErrStat,ErrMsg,RoutineName)
+      CALL SeaStOut_WrSummaryFile(InitInp, InputFileData, p, ErrStat2, ErrMsg2); if(Failed()) return;
 
       
 
       ! Setup the 4D grid information for the Interpolation Module
-      SeaSt_Interp_InitInp%n        = (/p%WaveField%NStepWave,p%nGrid(1),p%nGrid(2),p%nGrid(3)/)
-      SeaSt_Interp_InitInp%delta    = (/real(p%WaveDT,ReKi),p%deltaGrid(1),p%deltaGrid(2),p%deltaGrid(3)/)
-      SeaSt_Interp_InitInp%pZero(1) = 0.0  !Time
-      SeaSt_Interp_InitInp%pZero(2) = -InputFileData%X_HalfWidth
-      SeaSt_Interp_InitInp%pZero(3) = -InputFileData%Y_HalfWidth
-      SeaSt_Interp_InitInp%pZero(4) = -InputFileData%Z_Depth  ! zi
-      SeaSt_Interp_InitInp%Z_Depth  =  InputFileData%Z_Depth
-      call SeaSt_Interp_Init(SeaSt_Interp_InitInp, p%WaveField%seast_interp_p,  ErrStat2, ErrMsg2)
-         CALL SetErrStat(ErrStat2,ErrMsg2,ErrStat,ErrMsg,RoutineName)
+      WaveField_InitInp%n        = (/p%WaveField%NStepWave,p%nGrid(1),p%nGrid(2),p%nGrid(3)/)
```

Comment:
Related to my other comment... it seems like it would be easier to just set `p%WaveField%GridParams%*` here instead of making them `WaveField_InitInp%*` and then copying them over in the `WaveField_SetParam` routine. I've debated removing this with the SeaSt_Interp module many times in the past :-).

Reply:
I agree. I'm going to remove the `InitInp` type entirely and remove the init routine.  They are just extra code that isn't needed.

Reply:
Done!